cli/pkg/local/git.go

```diff
@@ -0,0 +1,101 @@
+package local
+
+import (
+	"context"
+	"errors"
+
+	"connectrpc.com/connect"
+	"github.com/rilldata/rill/cli/pkg/gitutil"
+	localv1 "github.com/rilldata/rill/proto/gen/rill/local/v1"
+	"go.uber.org/zap"
+)
+
+func (s *Server) WatchGitStatus(ctx context.Context, r *connect.Request[localv1.WatchGitStatusRequest], stream *connect.ServerStream[localv1.WatchGitStatusResponse]) error {
+	// Get authenticated admin client
+	if !s.app.ch.IsAuthenticated() {
+		return errors.New("must authenticate before performing this action")
+	}
+
+	project, err := s.app.ch.LoadProject(ctx, s.app.ProjectPath)
+	if err != nil {
+		return err
+	}
+
+	config, err := s.app.ch.GitHelper(project.OrgName, project.Name, s.app.ProjectPath).GitConfig(ctx)
+	if err != nil {
+		return err
+	}
+	remote, err := config.FullyQualifiedRemote()
+	if err != nil {
+		return err
+	}
+
+	err = gitutil.GitFetch(ctx, s.app.ProjectPath, remote)
+	if err != nil {
+		return err
+	}
+
+	return gitutil.PollGitStatus(ctx, s.app.ProjectPath, remote, func(gs gitutil.GitStatus) {
```

Comment:
I'm not sure I like the streaming API, but if necessary, then can we move all polling to the API handler itself? Since the logic here doesn't seem stateful, it doesn't seem necessary to push the polling logic into nested libraries.